enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/OnboardingService.java

```diff
@@ -228,6 +227,26 @@ public Response performCleanup(OnboardingCleanupRequest request) throws Onboardi
         return new Response();
     }
 
+    /**
+     * Verify process identifier.
+     * @param ownerId Owner identification.
+     * @param processId Process identifier from request.
+     * @throws OnboardingProcessException Thrown in case process identifier is invalid.
+     */
+    public void verifyProcessId(OwnerId ownerId, String processId) throws OnboardingProcessException {
+        Optional<OnboardingProcessEntity> processOptional = onboardingProcessRepository.findProcessByActivationId(ownerId.getActivationId());
+        if (!processOptional.isPresent()) {
+            logger.error("Onboarding process not found, {}", ownerId);
+            throw new OnboardingProcessException();
+        }
+        String expectedProcessId = processOptional.get().getId();
+
+        if (!expectedProcessId.equals(processId)) {
+            logger.warn("Invalid process ID received in request: {}", processId);
```

Comment:
Maybe we should log also the ownerId context here? But this will be solved also later by logging with MDC.

Reply:
Yes, good point, I will change it.